src/components/Contacts/ContactDetails/ContactDetailContext.tsx

```diff
@@ -68,6 +68,12 @@ export const ContactDetailProvider: React.FC<Props> = ({ children }) => {
     setSelectedTabKey(newKey);
   };
 
+  useEffect(() => {
+    if (query?.tabKey) {
+      setSelectedTabKey(query.tabKey);
```

Comment:
Oh, I see that in the useState it's `query.tab` and here it's `query.tabKey`. It would be great if we could use the same query param, but you're right that this isn't a logic error like I thought it was.

Reply:
Possibly. How you had it before, are the possible values for `query.tab` and `query.tabKey` the same? If they weren't, you'll need to do some kind of translation to ensure that all places that the places that set `query.tab` are setting it to values that the code that reads it expect.

Reply:
That makes sense. I think your refactor is good then! I didn't look at the code enough to notice that the `TabKey` lookup is a no-op (i.e. `TabKey[key] === key`).